### PR TITLE
feat(sso): login precheck

### PR DESCRIPTION
**Pull request by paolodamico**
## Problem

Part of the SAML / SSO enforcement saga.

## Changes

- We now do a precheck to see if there's an SSO enforcement policy for the domain of the user's email address. If there is, we redirect automatically. If there's not, we show the login page normally.

**How this looks like in a very, very, very slow connection**
image

**How this looks like in practice**
image

**How this looks like if you have SSO enforcement enabled**

image

image

## How did you test this code?
- Add an `OrganizationDomain` with `sso_enforcement` and see redirection. Try any other email address not on that domain and the password input will appear. Remember to add `sso_enfocement` to your available features or set a valid license.
- Django functional tests.

### CODE REVIEW

File: frontend/src/scenes/authentication/loginLogic.ts [hunk 1]
```diff
@@ -43,6 +54,11 @@ export const loginLogic = kea<loginLogicType<AuthenticateResponseType>>({
                 window.location.href = afterLoginRedirect()
             }
         },
+        precheckSuccess: ({ precheckResponse }) => {
+            if (precheckResponse?.sso_enforcement) {
+                window.location.href = `/login/${precheckResponse.sso_enforcement}/`
```
Comment (paolodamico) [inline frontend/src/scenes/authentication/loginLogic.ts:59]: Not sure about this UX @clarkus, is redirecting automatically okay or should we change the "Login" button to a "Login with X" and let the user click it?

File: posthog/api/authentication.py [hunk 2]
```diff
@@ -74,6 +74,26 @@ def create(self, validated_data: Dict[str, str]) -> Any:
...
+    email = serializers.EmailField()
+
+    def to_representation(self, instance: Dict[str, str]) -> Dict[str, Any]:
+        return instance
+
+    def create(self, validated_data: Dict[str, str]) -> Any:
+        email = validated_data.get("email")
+        domain = email[email.index("@") + 1 :]
+        domain_query = (
+            OrganizationDomain.objects.verified_domains()
```
Comment (paolodamico) [inline posthog/api/authentication.py:87]: FYI @guidoiaquinti here's the logic we discussed where we only use the domain to check (regardless of whether there's an account or not).

File: frontend/src/scenes/authentication/loginLogic.ts (hunk 1 above)
Comment (liyiy) [reply frontend/src/scenes/authentication/loginLogic.ts:59]: Right now it feels quite abrupt to go from one login to the SSO one, perhaps a transition here or at least an indication that the user is now logging in with SSO could help?

Comment (paolodamico) [reply frontend/src/scenes/authentication/loginLogic.ts]: Makes sense @liyiy, thanks for the feedback! Please see updated description with details.

**Review by mariusandra — APPROVED** [review]
Basically looks good, yet found some comments

File: frontend/src/scenes/authentication/loginLogic.ts (hunk 1 above)
Comment (liyiy) [reply frontend/src/scenes/authentication/loginLogic.ts]: lovely

**Review by liyiy — APPROVED** [review]

**Comment by paolodamico** [comment]
Thanks for the review @liyiy & @mariusandra, feedback addressed

**Pull request merged**